core/generator/generator.go

```diff
@@ -59,6 +59,17 @@ func New(
 	}
 }
 
+// PendingTxs returns all of the pendings txs that will be
+// included in the generator's next block.
+func (g *Generator) PendingTxs() []*bc.Tx {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+
+	txs := make([]*bc.Tx, len(g.pool))
```

Comment:
Since `g.pool` can still be modified (and even sorted in topo order), I think this needs to be an explicit copy of the slice.